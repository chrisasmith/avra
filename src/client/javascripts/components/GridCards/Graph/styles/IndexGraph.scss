@import '../../../../../stylesheets/modules/theme.scss';

.container {
  display: flex;
  position: relative;
  margin-bottom: 14px;
  align-items: center;

  &:last-child {
    margin-bottom: 0;
  }
}

.graphWrapper {
  width: calc(100% - 35px);
  white-space: nowrap;
  position: relative;
}

.title {
  width: 116px;
  font-size: 15px;
  font-size: 14px;
  line-height: 14px;
  padding-right: 4px;
}

.graph {
  flex-grow: 1;
}

.graphLine {
  background-color: #06b92d;
  height: 15px;
  margin-top: 0px;
  display: inline-block;

  &.first {
    background-color: #d3dcf6;
    left: 0;
    position: absolute;
  }
}

.graphAmount {
  font-size: 14px;
  line-height: 15px;
  vertical-align: top;
  margin-left: 8px;
  display: inline-block;

  &.first {
    font-weight: bold;
  }
}

.lineContainer {
  height: 15px;
  line-height: 15px;
}

.tooltip {
  border: 1px solid #c0c0c0;
    &:after {
      border-top-color: #000 !important;
      border-top-style: solid !important;
      border-top-width: 6px !important;
    }
}
