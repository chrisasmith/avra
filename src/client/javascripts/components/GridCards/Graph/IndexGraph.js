import React, { Component } from 'react'
import classnames from 'classnames'
import PropTypes from 'prop-types'
import ReactTooltip from 'react-tooltip'
import styles from './styles/IndexGraph.scss'

const propTypes = {
  title: PropTypes.string.isRequired,
  tooltip: PropTypes.string,
  value: PropTypes.number,
  maxValue: PropTypes.number,
}

class IndexGraph extends Component {
  constructor(props, context) {
    super(props, context)

    this.line = React.createRef()

    this.state = {
      expanded: false,
    }
  }

  render() {
    const { title, tooltip, value, maxValue } = this.props

    const percent = !maxValue ? value : (value * 100) / maxValue
    const midWay = 100 / maxValue

    return (
      <div className={styles.container}>
        <ReactTooltip type="info" />
        <div className={styles.title}>{title}</div>
        <div className={styles.graph}>
          <div className={styles.lineContainer}>
            <div>
              <div className={styles.graphWrapper} data-tip={tooltip} ref={this.line}>
                <div
                  style={{ width: `${percent}%` }}
                  className={classnames(styles.graphLine)}
                />
                <div
                  style={{ width: `${percent}%`, maxWidth: `${midWay}%` }}
                  className={classnames(styles.graphLine, styles.first)}
                />
                <div
                  className={classnames(styles.graphAmount, styles.first)}
                >
                  {value.toFixed(2)}
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    )
  }
}

IndexGraph.propTypes = propTypes
export default IndexGraph
