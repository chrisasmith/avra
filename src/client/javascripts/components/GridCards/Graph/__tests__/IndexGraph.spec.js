import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import IndexGraph from '../IndexGraph'

describe('IndexGraph Component', () => {
  it('Should show', () => {
    const wrapper = shallow(<IndexGraph
      title="title"
      tooltip="tip"
      value={1.1}
      maxValue={2}
    />)

    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('Should show less than 0', () => {
    const wrapper = shallow(<IndexGraph
      title="title"
      tooltip="tip"
      value={0.2}
      maxValue={2}
    />)

    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('Should show with a large max', () => {
    const wrapper = shallow(<IndexGraph
      title="title"
      tooltip="tip"
      value={0.2}
      maxValue={6}
    />)

    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
