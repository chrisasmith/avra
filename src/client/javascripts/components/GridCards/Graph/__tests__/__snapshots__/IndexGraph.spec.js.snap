// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexGraph Component Should show 1`] = `
<div
  className="container"
>
  <ReactTooltip
    insecure={true}
    resizeHide={true}
    type="info"
    wrapper="div"
  />
  <div
    className="title"
  >
    title
  </div>
  <div
    className="graph"
  >
    <div
      className="lineContainer"
    >
      <div>
        <div
          className="graphWrapper"
          data-tip="tip"
        >
          <div
            className="graphLine"
            style={
              Object {
                "width": "55.00000000000001%",
              }
            }
          />
          <div
            className="graphLine first"
            style={
              Object {
                "maxWidth": "50%",
                "width": "55.00000000000001%",
              }
            }
          />
          <div
            className="graphAmount first"
          >
            1.10
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`IndexGraph Component Should show less than 0 1`] = `
<div
  className="container"
>
  <ReactTooltip
    insecure={true}
    resizeHide={true}
    type="info"
    wrapper="div"
  />
  <div
    className="title"
  >
    title
  </div>
  <div
    className="graph"
  >
    <div
      className="lineContainer"
    >
      <div>
        <div
          className="graphWrapper"
          data-tip="tip"
        >
          <div
            className="graphLine"
            style={
              Object {
                "width": "10%",
              }
            }
          />
          <div
            className="graphLine first"
            style={
              Object {
                "maxWidth": "50%",
                "width": "10%",
              }
            }
          />
          <div
            className="graphAmount first"
          >
            0.20
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`IndexGraph Component Should show with a large max 1`] = `
<div
  className="container"
>
  <ReactTooltip
    insecure={true}
    resizeHide={true}
    type="info"
    wrapper="div"
  />
  <div
    className="title"
  >
    title
  </div>
  <div
    className="graph"
  >
    <div
      className="lineContainer"
    >
      <div>
        <div
          className="graphWrapper"
          data-tip="tip"
        >
          <div
            className="graphLine"
            style={
              Object {
                "width": "3.3333333333333335%",
              }
            }
          />
          <div
            className="graphLine first"
            style={
              Object {
                "maxWidth": "16.666666666666668%",
                "width": "3.3333333333333335%",
              }
            }
          />
          <div
            className="graphAmount first"
          >
            0.20
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
